coding/javascript.md

```diff
@@ -3,33 +3,93 @@ layout: base
-- Understand when you cause an element repaint and reduce inline style manipulations accordingly.
-- If possible, always use CSS for element transforms and transitions.
+### Dependency management
+We use [yarn](https://yarnpkg.com/en) for dependency management.
+
+### Testing
+We use the [jest](https://jestjs.io) unittest framework.
+
+### Code style & formatting
+To ensure code formatting consistency across our codebase, we use [Prettier](https://github.com/prettier/prettier).
+
+JS should also be written using [AirBnB](https://github.com/airbnb/javascript) style. New projects can adhere to this by selecting 'AirBnB' during `eslint init`, and existing projects can find the relevant eslint config in [eslint-config-airbnb](https://www.npmjs.com/package/eslint-config-airbnb).
+
+## React
+
+### Starting a new project
+When starting a new react application, we recommend [create-react-app](https://github.com/facebook/create-react-app) for bootstrapping the project. Avoid [ejecting](https://facebook.github.io/create-react-app/docs/available-scripts#npm-run-eject) as long as feasible to make updating dependencies easier.
+
+### Preferred libraries
+All projects should generally use the following:
+
+* style framework - [vanilla](https://github.com/vanilla-framework/vanilla-framework)
+* component testing - [enzyme](https://github.com/airbnb/enzyme)
+
+If you require routing, or state management:
+
+* routing - [react-router](https://github.com/ReactTraining/react-router)
+* app state management - [redux](https://redux.js.org)
+
+#### Redux specific
+* selector composition and memoisation - [reselect](https://github.com/reduxjs/reselect)
+* async/side-effects management (http, localstorage etc.) - [redux-saga](https://github.com/redux-saga/redux-saga)
+
+If you'd like introduce a new library, or feel we should replace one of the above, please make a PR to start a discussion.
+
+### Reference projects
+[CRBS-UI](https://git.launchpad.net/~crbs/crbs/+git/crbs-ui/tree/), while under active development, probably best reflects our standards for react projects currently.
+
+[build.snapcraft.io](https://github.com/canonical-websites/build.snapcraft.io) - a bit dated, but first fully React centred project in the webteam. Our first use of React, server-side rendering, Redux, Enzyme, etc. While it doesn't fully follow our current standards it may be a good place to check how some of these libraries were used.
+
+[canonical-webteam/react-starter](https://github.com/canonical-webteam/react-starter) was created to provide a codified example of our standards for react projects (Prettier, AirBnB style). While potentially still a useful reference, it should mostly be considered deprecated as it may not accurately reflect some of the choices we've made in real-world projects.
+
+### File naming conventions
+
+* Component files should use CamelCase and match the name of the default export (e.g. MyComponent.js).
```

Comment:
It seems like it's quite common both ways 🤷‍♂️

Reply:
Yep, in this case we're just conforming to the [AirBnB style guide for react](https://github.com/airbnb/javascript/blob/master/react/README.md), but I can see why it might feel a little weird, particularly for a linux user.

Reply:
I think CamelCase is pretty standard for React component files